ci/travis_before_script_cpp.sh

```diff
@@ -31,7 +31,8 @@ CPP_DIR=$TRAVIS_BUILD_DIR/cpp
 
 CMAKE_COMMON_FLAGS="\
 -DARROW_BUILD_BENCHMARKS=ON \
--DCMAKE_INSTALL_PREFIX=$ARROW_CPP_INSTALL"
+-DCMAKE_INSTALL_PREFIX=$ARROW_CPP_INSTALL \
+-DCMAKE_CXX_STANDARD=11"
```

Comment:
OK, we should either 

a) set this variable inside our CMakeLists.txt so it doesn't have to be specified on the command line
b) rely on the existing `-std=c++11` being set in the CXX_COMMON_FLAGS https://github.com/apache/arrow/blob/master/cpp/cmake_modules/SetupCxxFlags.cmake#L47

but not both